### ISSUE TITLE
nexus tests fail due to missing new dependency matplotlib

### ISSUE THREAD

**Issue by prckent** [QMCPACK/qmcpack#2273]
https://cdash.qmcpack.org/CDash/viewTest.php?onlydelta&buildid=103138

This optional dependency has become a requirement.

**Comment by jtkrogel** [comment]
That was unintentional.  I'll put in a fix shortly.  Do we have CI for "only required dependencies" and for "all dependencies"?

**Comment by prckent** [comment]
No.

I would be helpful to have a maintained list of the dependencies, both for Nexus and QMCPACK. e.g. WIthout lxml the pyscf spline conversion test can not work.

**Comment by grahamlopez** [comment]
Ideally, both such lists would be implicit in and enforced by the build system - and clear to find/read by humans.

**Comment by jtkrogel** [comment]
Nexus maintains its own list.  It is printed each time it is run by a user:

```
_____________________________________________________

                     Nexus 2.0.0

        (c) Copyright 2012-  Nexus developers

                     Please cite:
  J. T. Krogel Comput. Phys. Commun. 198 154 (2016)
     https://doi.org/10.1016/j.cpc.2015.08.012
_____________________________________________________
          
            

Checking for Nexus dependencies on the current machine...

  Nexus dependencies available on current machine:
    python3      = 3.6.9       (required)
    numpy        = 1.17.4      (required)
    scipy        = 0.19.1      (optional)
    h5py         = 2.7.1       (optional)
    matplotlib   = 2.1.1       (optional)
    pydot        = 1.2.3       (optional)
    spglib       = 1.14.1      (optional)
    seekpath     = 1.9.3       (optional)
    pycifrw      = (unknown)   (optional)
    cif2cell     = (unknown)   (optional)
```

**Comment by grahamlopez** [comment]
nice. we could do similar for qmcpack in the cmake.

**Comment by prckent** [comment]
@jtkrogel Good. Can you please update the manual & readme to mention this and list these dependencies? somewhere in installation.tex

**Comment by jtkrogel** [comment]
The dependencies are described in section 3.2 of the Nexus user guide.  The Nexus README hasn't been updated for a while (Python 2 still discussed!).  I will make an update there.

If you want, I can repeat this information in the QMCPACK manual and readme, but I am leery of maintaining 4x duplicated information.  Possibly during the move to ReadTheDocs for both QMCPACK and Nexus, we can make a better coupling (via links) between the two manuals.  Let me know.

**Comment by jtkrogel** [comment]
See #2276.

**Comment by prckent** [comment]
I agree on avoiding duplication. We could add a link to the NEXUS manual PDF today https://docs.qmcpack.org/nexus_user_guide.pdf

**Comment by prckent** [comment]
Closed by #2274 

Confirmed working https://cdash.qmcpack.org/CDash/buildSummary.php?buildid=103255
